.jenkins/infrastructure/docker/build_mystikos_docker_images.Jenkinsfile

```diff
@@ -0,0 +1,139 @@
+        booleanParam(name: "TAG_LATEST", defaultValue: false, description: "Update the latest tag to the currently built DOCKER_TAG")
+    }
+    environment {
+        INTERNAL_REPO_CREDS = 'mystikos-internal-container-registry'
+        // Docker plugin cannot seem to use credentials from Azure Key Vault
+        DOCKERHUB_REPO_CREDS = ''
+        BASE_DOCKERFILE_DIR = ".jenkins/docker/base/"
+    }
+    stages {
+        stage("Checkout") {
+            steps {
+                cleanWs()
+                checkout([$class: 'GitSCM',
+                    branches: [[name: BRANCH_NAME]],
+                    extensions: [],
+                    userRemoteConfigs: [[url: "https://github.com/${params.REPOSITORY_NAME}"]]])
+            }
+        }
+        stage("Base Image") {
+            stages {
+                stage('Build base container') {
+                    steps {
+                        dir(env.BASE_DOCKERFILE_DIR) {
+                            sh """
+                                chmod +x ./build.sh
+                                mkdir build
+                                cd build
+                                ../build.sh -m "${params.MYST_VERSION}" -o "${params.OE_BASE_CONTAINER_TAG}" -u "18.04" -t "${params.MYST_BASE_DOCKER_TAG}"
+                                ../build.sh -m "${params.MYST_VERSION}" -o "${params.OE_BASE_CONTAINER_TAG}" -u "20.04" -t "${params.MYST_BASE_DOCKER_TAG}"
+                            """
+                        }
+                    }
+                }
+                stage('Push base containers to internal repository') {
+                    when {
+                        expression { return params.PUBLISH_CONTAINERS }
+                    }
+                    steps {
+                        script {
+                            tag_bionic = "mystikos-bionic:${params.MYST_BASE_DOCKER_TAG}"
+                            tag_focal = "mystikos-focal:${params.MYST_BASE_DOCKER_TAG}"
+                            docker.withRegistry(params.INTERNAL_REPO, env.INTERNAL_REPO_CREDS) {
+                                base_1804_image = docker.image(tag_bionic)
+                                base_2004_image = docker.image(tag_focal)
+
+                                base_1804_image.push()
+                                base_2004_image.push()
+                                if ( params.TAG_LATEST ) {
+                                    base_1804_image.push('latest')
+                                    base_2004_image.push('latest')
+                                }
+                            }
+                            sh "docker logout"
+                        }
+                    }
+                }
+                stage('Build container requirements') {
+                    steps {
+                        script {
+                            containers_reqs.each { container -> 
```

Comment:
How readable is the output from this stage? I am wondering how useful it would be to add some sort of visual delimiter and/or summary at the end of the stage in case there are multiple failures per build

Reply:
I don't have any qualms with the readability: https://oe-jenkins-dev.westeurope.cloudapp.azure.com/blue/organizations/jenkins/Private%2FCICD_Infrastructure%2FMystikos-Base-Docker-Container/detail/Mystikos-Base-Docker-Container/36/pipeline/

what do you think?

Reply:
Actually that looks great!